### PR TITLE
diff: some fix

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 

1. fix issue #354 
  - diff will update and print summary info every 10 seconds, and the summary info is calculated by select `chunk` table.
  - diff will save chunk's information to `chunk` table after all chunk is split.  

  when update summary info, the table may not finish split chunk, so there is no data in `chunk` table, and diff will print warn log, users may be confused by this.

2. update parser and tidb by the way, and a minor fix on set `AllowAutoRandom` to support parser `auto random`

3. support table name and column name with ``` 

### What is changed and how it works?

1.  just print a debug log and don't return error for `getChunkSummary`
2. update parser and tidb version
3.  replace ``column name`` with `dbutil.ColumnName(column name)`

### CODE REVIEW

**Review by lichunzhu — DISMISSED** [review]
LGTM

**Review by july2993 — DISMISSED** [review]

File: pkg/diff/diff_test.go [hunk 1]
```diff
@@ -48,50 +48,52 @@ func (*testDiffSuite) TestGenerateSQLs(c *C) {
...
-	c.Assert(replaceSQL, Equals, "REPLACE INTO `test`.`atest`(`id`,`name`,`birthday`,`update_time`,`money`) VALUES (NULL,'a\\'a','2018-01-01 00:00:00','10:10:10',11.1111);")
-	c.Assert(deleteSQL, Equals, "DELETE FROM `test`.`atest` WHERE `id` is NULL AND `name` = 'a\\'a' AND `birthday` = '2018-01-01 00:00:00' AND `update_time` = '10:10:10' AND `money` = 11.1111;")
+	replaceSQL = generateDML("replace", rowsData, tableInfo, "diff_test")
+	deleteSQL = generateDML("delete", rowsData, tableInfo, "diff_test")
+	c.Assert(replaceSQL, Equals, "REPLACE INTO `diff_test`.`atest`(`id`,`name`,`birthday`,`update_time`,`money`) VALUES (NULL,'a\\'a','2018-01-01 00:00:00','10:10:10',11.1111);")
+	c.Assert(deleteSQL, Equals, "DELETE FROM `diff_test`.`atest` WHERE `id` is NULL AND `name` = 'a\\'a' AND `birthday` = '2018-01-01 00:00:00' AND `update_time` = '10:10:10' AND `money` = 11.1111;")
 }
 
-func (t *testDiffSuite) testDiff(c *C) {
+func (t *testDiffSuite) TestDiff(c *C) {
```
Comment (WangXiangUSTC) [inline pkg/diff/diff_test.go:86]: this function is not run before 🤢

Comment (july2993) [reply pkg/diff/diff_test.go:86]: Screen Shot 2020-06-18 at 3 31 50 PM

suggest use golangci-lint to help improve the code

Comment (WangXiangUSTC) [reply pkg/diff/diff_test.go:86]: 👍

Comment (WangXiangUSTC) [reply pkg/diff/diff_test.go:86]: I will update this in next pr

File: pkg/diff/diff.go [hunk 2]
```diff
@@ -90,7 +89,7 @@ type TableDiff struct {
 	IgnoreDataCheck bool `json:"-"`
 
 	// set true will continue check from the latest checkpoint
-	UseCheckpoint bool `json:"use-checkpoint"`
+	UseCheckpoint bool `json:"-"`
```
Comment (WangXiangUSTC) [inline pkg/diff/diff.go:92]: fix issue https://github.com/pingcap/tidb-tools/issues/358

**Review by lichunzhu — APPROVED** [review]
LGTM

**Pull request merged**